src/main/scala/is/hail/expr/AST.scala

```diff
@@ -574,6 +588,15 @@ case class ApplyMethod(posn: Position, lhs: AST, method: String, args: Array[AST
     lhs.typecheck(ec)
     (lhs.`type`, method, args) match {
 
+      case (it: TAggregable, "groupBy", Array(Lambda(_, param1, body1), Lambda(_, param2, body2), rest@_*)) =>
+        rest.foreach(_.typecheck(ec.copy(st = emptySymTab)))
+        body1.typecheck(ec.copy(st = it.symTab + ((param1, (-1, it.elementType)))))
+        body2.typecheck(ec.copy(st = ec.st.filter { case (name, (idx, typ)) => !typ.isInstanceOf[TAggregable] } + ((param2, (it.symTab.find { case (_, (_, t)) => t == it.elementType }.get._2._1, it)))))
```

Comment:
As per our discussion, we need to allow for multiple aggregations inside the second expression in the groupBy.